### PR TITLE
Code refactoring.

### PR DESCRIPTION
**Pull request by patrycja93**
Changed checking input from user to UI class because Match class
shouldn't be responsible for it. Deleted checking if match is finished
in GameSetting class. Corrected documentation.

### CODE REVIEW

File: src/main/java/pl/patrycja/ox/board/Board.java [hunk 1]
```diff
@@ -4,23 +4,21 @@
...
+     * Called whenever we want to add a sign to the board.
      *
      * @param fieldNumber is a number of field in board where we want to add a sign
      * @param sign        an argument which will be added to board
      */
     boolean putSignToBoard(int fieldNumber, Sign sign);
 
     /**
-     * This method is called whenever we want to delete from the board all signs.
+     * Called whenever we want to delete from the board all signs.
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/board/Board.java]: Maybe `Called whenever we want to delete all signs from the board`?

File: src/main/java/pl/patrycja/ox/gameflow/Game.java [hunk 2]
```diff
@@ -11,40 +11,33 @@
...
     }
 
     public void play() {
-        showDemoText(ifOpenAutomaticTests);
         gameSettings.setPlayer();
-        List<Spectator> spectators = SpectatorsRoom.addSpectators(gameSettings);
         PlayerChanger playerChanger = new PlayerChanger(List.of(firstPlayer, secondPlayer), gameSettings);
 
-        while (!gameSettings.isEndGame()) {
+        for (int i = 0; i < 3; i++) {
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/Game.java]: Why 3?

File: src/main/java/pl/patrycja/ox/ui/InputChecker.java [hunk 3]
```diff
@@ -7,13 +7,30 @@
...
+        return Integer.parseInt(String.valueOf(input));
+    }
+
     /**
      * This method is called to check if number is Integer. It returns true
      * in case it is an Integer, otherwise returns false.
      *
      * @param number is an argument to check if it is an Integer
      */
-    public static boolean checkNumber(String number, UI ui) {
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/ui/InputChecker.java:16]: Do you still need this documentation if method is private?

File: src/main/java/pl/patrycja/ox/ui/InputChecker.java [hunk 4]
```diff
@@ -7,13 +7,30 @@
...
 
+    private UI ui;
+    private int boardSize;
+
+    public InputChecker(UI ui, int boardSize) {
+        this.ui = ui;
+        this.boardSize = boardSize;
+    }
+
+    public int getNumber() {
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/ui/InputChecker.java]: Missing documentation.

**Review by BulaMar — APPROVED** [review]
Not Bad, only smal comments.

File: src/main/java/pl/patrycja/ox/board/Board.java (hunk 1 above)
Comment (patrycja93) [reply src/main/java/pl/patrycja/ox/board/Board.java]: ok

File: src/main/java/pl/patrycja/ox/gameflow/Game.java (hunk 2 above)
Comment (patrycja93) [reply src/main/java/pl/patrycja/ox/gameflow/Game.java]: done.

File: src/main/java/pl/patrycja/ox/ui/InputChecker.java (hunk 3 above)
Comment (patrycja93) [reply src/main/java/pl/patrycja/ox/ui/InputChecker.java:16]: I will delete it.

File: src/main/java/pl/patrycja/ox/ui/InputChecker.java (hunk 4 above)
Comment (patrycja93) [reply src/main/java/pl/patrycja/ox/ui/InputChecker.java]: I will add it.

File: src/test/java/pl/patrycja/ox/TestUI.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package pl.patrycja.ox;
+
+import pl.patrycja.ox.ui.UI;
+
+public class TestUI implements UI {
```
Comment (BulaMar) [inline src/test/java/pl/patrycja/ox/TestUI.java:5]: 👍

File: src/main/java/pl/patrycja/ox/GameSettings.java [hunk 6]
```diff
@@ -14,7 +14,7 @@
     private boolean endMatch = false;
     private Sign firstPlayer;
     private int unbrokenLine;
-    private int matchNumber;
+    private int numberOfMatches;
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/GameSettings.java:17]: 👍

**Review by BulaMar — APPROVED** [review]
Thanks!
Good that you fixed those broken test cases 😉

**Pull request merged**